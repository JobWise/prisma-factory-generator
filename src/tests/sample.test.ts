import { getDMMF } from '@prisma/sdk'
import {
  addModelFactoryDeclaration,
  addPrismaImportDeclaration,
} from '../generator'
import { Project, ts } from 'ts-morph'
const datamodel = /* Prisma */ `
model User {
  id          Int          @id @default(autoincrement())
  email       String       @unique
  createdAt   DateTime     @default(now()) @map(name: "created_at")
  updatedAt   DateTime     @default(now()) @map(name: "updated_at")
  accessToken AccessToken?
  uuid        String       @unique @default(uuid())

  @@map(name: "users")
}

model AccessToken {
  id         Int      @id @default(autoincrement())
  user       User     @relation(fields: [userId], references: [id])
  userId     Int      @unique @map(name: "user_id")
  createdAt  DateTime @default(now()) @map(name: "created_at")
  paidAmount Int      @map(name: "paid_amount")
  isActive   Boolean  @default(false)

  @@map("access_tokens")
}
`

test('generate model', async () => {
  const project = new Project({})
  const sourceFile = project.createSourceFile('./src/test-src.ts')
  const dmmf = await getDMMF({ datamodel })
  addPrismaImportDeclaration(sourceFile)
  dmmf.datamodel.models.forEach((model) => {
    addModelFactoryDeclaration(sourceFile, model)
  })

  sourceFile.formatText({
    indentSize: 2,
    semicolons: ts.SemicolonPreference.Remove,
  })
  expect(sourceFile.getFullText()).toMatchSnapshot()
})
