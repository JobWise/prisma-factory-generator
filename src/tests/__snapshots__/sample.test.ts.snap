// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate model 1`] = `
"import { PrismaClient } from \\"@prisma/client\\"
import { Prisma } from \\".prisma/client\\"
const prisma = new PrismaClient()

export const UserDefaultVariables = {
  id: faker.datatype.number(),
  email: faker.name.title(),
  createdAt: faker.datatype.datetime(),
  updatedAt: faker.datatype.datetime(),
  uuid: faker.name.title()
}

export async function createUser(args: Prisma.UserCreateArgs) {

  return await prisma.User.create({
    data: {
      ...UserDefaultVariables,
      ...args.data,
    }
  })

}


export const AccessTokenDefaultVariables = {
  id: faker.datatype.number(),
  userId: faker.datatype.number(),
  createdAt: faker.datatype.datetime(),
  paidAmount: faker.datatype.number(),
  isActive: false
}

export async function createAccessToken(args: Prisma.AccessTokenCreateArgs) {

  return await prisma.AccessToken.create({
    data: {
      ...AccessTokenDefaultVariables,
      ...args.data,
    }
  })

}
"
`;
